backend/frontend/views.py

```diff
@@ -368,16 +377,63 @@ def action_start_card(request, card_id):
 
 
 @user_passes_test_or_forbidden(can_view_user_board)
-def course_component_details(request, project_id):
-    project = get_object_or_404(RecruitProject, id=project_id)
+def topic(request, id):
+    course_component = get_object_or_404(TopicProgress, id=id)
+
+    log_entries = LogEntry.objects.filter(object_1_id=id)
+
+    relevant_logs = []
+    for log_entry in log_entries:
+        event_type = EventType.objects.get(id=log_entry.event_type_id)
+
+        if (
+            course_component.agile_card.status == AgileCard.IN_PROGRESS
+            and event_type.name == CARD_STARTED
+        ) or (
+            course_component.agile_card.status == AgileCard.IN_PROGRESS
+            and event_type.name == CARD_REVIEW_REQUEST_CANCELLED
+        ):
```

Comment:
More things...

- This if check should be rewritten so that we don't have to the following twice:
```
course_component.agile_card.status == AgileCard.IN_PROGRESS
```

- I feel like there should be a way to just filter LogEntries without needing to use if checks or event types. You just need eventtypes that match `card started` and `card review request cancelled`. My brain might just be tired so maybe not, just think about it or have a closer look.

Reply:
@Ngoakor12, here you mean something like the code in the image below, right?

![Screenshot 2024-03-27 084436](https://github.com/Umuzi-org/Tilde/assets/94048861/92b6c65f-0779-41d5-9db8-d017e58707b6)

Reply:
Yeah, but maybe let's leave the sorting out for now and maybe combine some things. See:
```py
relevant_logs = LogEntry.objects.filter((Q(event_type__name=CARD_STARTED)|Q(event_type__name=CARD_REVIEW_REQUEST_CANCELLED)),object_1_id=self.id)

# then the sorting or just sort using while doing the query if possible
```

Reply:
I'm just freestyling, so please make sure whatever you come up with works and feel free to use any other approaches if you feel this is unreasonable

Reply:
@Raymond-Mawina, this is now done. ✅. I will move the log to dedicated property method and make the rest of the changes. thanks.